dashboard/lib/api/v1/school_district_autocomplete.rb

```diff
@@ -0,0 +1,38 @@
+class Api::V1::SchoolDistrictAutocomplete < AutocompleteHelper
+  # Queries the schools lookup table for schools that match the user-defined
+  # search criteria.
+  # @param query [String] the user-define query string
+  # @param limit [int] the maximum number of results to return
+  # @return [Array] an array of JSON formatted schools
+  def self.get_matches(query, limit)
+    limit = format_limit(limit)
+    query = format_query(query)
+
+    return [] if query.length < MIN_WORD_LENGTH + 2
+
+    rows = SchoolDistrict.limit(limit).
+      where("MATCH(name,city) AGAINST(? IN BOOLEAN MODE)", query).
+      order("MATCH(name,city) AGAINST('#{query}' IN BOOLEAN MODE) DESC, state, city, name")
+
+    return rows.map do |row|
+      Serializer.new(row).attributes
+    end
+  end
+
```

Comment:
The get_matches code for school_autocomplete and school_district_autocomplete is almost identical. Can it be moved into the base class?

Reply:
Almost but one does the zip code search.

Reply:
If that's the only difference, then perhaps it could still be common code with a parameter for the included search fields as a string array, which could be injected into the query with `?` and then `included_fields.join(',')`

Reply:
Well it's right in the middle. I think it would be preferable to see how the auto-complete evolves before refactoring and adding complexity around conditionals. I can take another look later today if I still have time as well to see if there is any more opportunity to refactor.